<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>celltraj.trajectory.Trajectory &mdash; celltraj 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=a58bc63e"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            celltraj
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../readme.html">celltraj</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api.html">API Reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">celltraj</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">celltraj.trajectory.Trajectory</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/stubs/celltraj.trajectory.Trajectory.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="celltraj-trajectory-trajectory">
<h1>celltraj.trajectory.Trajectory<a class="headerlink" href="#celltraj-trajectory-trajectory" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">celltraj.trajectory.</span></span><span class="sig-name descname"><span class="pre">Trajectory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5filename</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/celltraj/trajectory.html#Trajectory"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory" title="Link to this definition"></a></dt>
<dd><p>A toolset for single-cell trajectory modeling. See:</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This code, currently, should be considered as an untested pre-release version</p>
</div>
<p class="rubric">References</p>
<p>Copperman, Jeremy, Sean M. Gross, Young Hwan Chang, Laura M. Heiser, and Daniel M. Zuckerman.
“Morphodynamical cell state description via live-cell imaging trajectory embedding.”
Communications Biology 6, no. 1 (2023): 484.</p>
<p>Copperman, Jeremy, Ian C. Mclean, Sean M. Gross, Young Hwan Chang, Daniel M. Zuckerman, and Laura M. Heiser.
“Single-cell morphodynamical trajectories enable prediction of gene expression accompanying cell state change.”
bioRxiv (2024): 2024-01.</p>
<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5filename</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/celltraj/trajectory.html#Trajectory.__init__"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a Trajectory object, optionally loading metadata from an HDF5 file.</p>
<p>This constructor sets the HDF5 filename and attempts to load metadata associated with the file.
If the file is present, it reads the metadata from a predefined group. Errors during metadata
loading are caught and logged. Future updates should include better commenting and organizational
improvements of class attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>h5filename</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – The path to the HDF5 file from which to load the metadata. If not provided, the
instance will be initialized without loading metadata.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>TODO:
- Improve documentation of class attributes.
- Reorganize attributes into a more meaningful structure.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="go">loading path/to/your/hdf5file.h5</span>
</pre></div>
</div>
</dd></dl>

<p class="rubric">Methods</p>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.__init__" title="celltraj.trajectory.Trajectory.__init__"><code class="xref py py-obj docutils literal notranslate"><span class="pre">__init__</span></code></a>([h5filename])</p></td>
<td><p>Initialize a Trajectory object, optionally loading metadata from an HDF5 file.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_Xtraj_celltrajectory" title="celltraj.trajectory.Trajectory.get_Xtraj_celltrajectory"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_Xtraj_celltrajectory</span></code></a>(cell_traj[, Xtraj, ...])</p></td>
<td><p>Retrieves trajectory segments for a specific cell trajectory from a larger set of trajectory data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_alpha" title="celltraj.trajectory.Trajectory.get_alpha"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_alpha</span></code></a>(i1, i2)</p></td>
<td><p>Calculates the alignment measure, alpha, between two cells identified by their indices.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_beta" title="celltraj.trajectory.Trajectory.get_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_beta</span></code></a>(i1, i2)</p></td>
<td><p>Calculates the cosine of the angle (beta) between the movement directions of two cells.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_blocks" title="celltraj.trajectory.Trajectory.get_cell_blocks"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_blocks</span></code></a>(label)</p></td>
<td><p>Extracts bounding box information for each cell from a labeled mask image.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_channel_crosscorr" title="celltraj.trajectory.Trajectory.get_cell_channel_crosscorr"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_channel_crosscorr</span></code></a>([indcells, ...])</p></td>
<td><p>Computes the cross-correlation between two image channels within labeled cells, using masks defined by a specific mask channel.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_compartment_ratio" title="celltraj.trajectory.Trajectory.get_cell_compartment_ratio"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_compartment_ratio</span></code></a>([indcells, ...])</p></td>
<td><p>Calculates the ratio of features between two cellular compartments, optionally adjusted for image intensity and morphology transformations.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_data" title="celltraj.trajectory.Trajectory.get_cell_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_data</span></code></a>(ic[, frametype, ...])</p></td>
<td><p>Retrieves image and mask data for specific cells based on various configuration options.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_features" title="celltraj.trajectory.Trajectory.get_cell_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_features</span></code></a>(function_tuple[, ...])</p></td>
<td><p>Extracts complex cell features based on the specified custom functions and imaging data.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_index" title="celltraj.trajectory.Trajectory.get_cell_index"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_index</span></code></a>([verbose, save_h5, overwrite])</p></td>
<td><p>Computes indices and corresponding frame information for each cell in an image stack, capturing this data in several attributes.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_positions" title="celltraj.trajectory.Trajectory.get_cell_positions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_positions</span></code></a>([mskchannel, save_h5, ...])</p></td>
<td><p>Calculate the center of mass for cells in each frame of the mask channel and optionally save these positions to an HDF5 file.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_cell_trajectory" title="celltraj.trajectory.Trajectory.get_cell_trajectory"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_cell_trajectory</span></code></a>(cell_ind[, n_hist])</p></td>
<td><p>Retrieves the trajectory of a specified cell across previous frames, tracing back from the current frame to the point of its first appearance or until a specified number of history steps.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_dx" title="celltraj.trajectory.Trajectory.get_dx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_dx</span></code></a>(i1)</p></td>
<td><p>Calculates the displacement vector of a cell between its current position and its previous position in the trajectory.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_fmask_data" title="celltraj.trajectory.Trajectory.get_fmask_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_fmask_data</span></code></a>(n_frame[, channel])</p></td>
<td><p>Retrieve the foreground mask data for a specific frame using different methods depending on the set attributes.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_frames" title="celltraj.trajectory.Trajectory.get_frames"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_frames</span></code></a>()</p></td>
<td><p>Scans an HDF5 file for image and mask datasets to determine the total number of frames, images per frame, and the total number of cells across all frames.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_image_data" title="celltraj.trajectory.Trajectory.get_image_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_image_data</span></code></a>(n_frame)</p></td>
<td><p>Retrieve the image data for a specified frame from the HDF5 file associated with this instance.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_image_shape" title="celltraj.trajectory.Trajectory.get_image_shape"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_image_shape</span></code></a>([n_frame])</p></td>
<td><p>Determine the dimensions of the image and mask data from the HDF5 file at a specified frame index and store these as attributes.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_lineage_btrack" title="celltraj.trajectory.Trajectory.get_lineage_btrack"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_lineage_btrack</span></code></a>([mskchannel, distcut, ...])</p></td>
<td><p>Tracks cell lineages over an image stack using Bayesian tracking with visual confirmation options.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_lineage_mindist" title="celltraj.trajectory.Trajectory.get_lineage_mindist"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_lineage_mindist</span></code></a>([distcut, visual, ...])</p></td>
<td><p>Tracks cell lineage based on the minimum distance between cells across consecutive frames.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_mask_data" title="celltraj.trajectory.Trajectory.get_mask_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_mask_data</span></code></a>(n_frame)</p></td>
<td><p>Retrieve the mask data for a specified frame from the HDF5 file associated with this instance.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_motility_features" title="celltraj.trajectory.Trajectory.get_motility_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_motility_features</span></code></a>([indcells, ...])</p></td>
<td><p>Extracts motility features for individual cells and their neighbors.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_pair_rdf" title="celltraj.trajectory.Trajectory.get_pair_rdf"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_pair_rdf</span></code></a>([cell_indsA, cell_indsB, ...])</p></td>
<td><p>Calculates the radial distribution function (RDF) between two sets of cells, identifying the frequency of cell-cell distances within specified radial bins.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_secreted_ligand_density" title="celltraj.trajectory.Trajectory.get_secreted_ligand_density"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_secreted_ligand_density</span></code></a>(frame[, ...])</p></td>
<td><p>Simulates the diffusion of secreted ligands from cells, providing a spatial distribution of ligand density across a specified frame.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_signal_contributions" title="celltraj.trajectory.Trajectory.get_signal_contributions"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_signal_contributions</span></code></a>(S[, time_lag, ...])</p></td>
<td><p>Computes the spatial contributions of signaling between cells over a specified time lag.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_stack_trans" title="celltraj.trajectory.Trajectory.get_stack_trans"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_stack_trans</span></code></a>([mskchannel, ntrans, maxt, ...])</p></td>
<td><p>Computes translations across an image stack using a brute force optimization method to align cell centers from frame to frame.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_trajAB_segments" title="celltraj.trajectory.Trajectory.get_trajAB_segments"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_trajAB_segments</span></code></a>(xt[, stateA, stateB, ...])</p></td>
<td><p>Identifies segments within trajectories that transition between specified states, A and B.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_traj_segments" title="celltraj.trajectory.Trajectory.get_traj_segments"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_traj_segments</span></code></a>(seg_length)</p></td>
<td><p>Divides each trajectory into multiple overlapping segments of a specified length.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_trajectory_steps" title="celltraj.trajectory.Trajectory.get_trajectory_steps"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_trajectory_steps</span></code></a>([inds, traj, Xtraj, ...])</p></td>
<td><p>Extracts sequential steps from cell trajectories and retrieves corresponding features from a feature matrix.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.get_unique_trajectories" title="celltraj.trajectory.Trajectory.get_unique_trajectories"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_unique_trajectories</span></code></a>([cell_inds, ...])</p></td>
<td><p>Computes unique trajectories for a set of cells over multiple frames, minimizing redundancy by ensuring that no two trajectories cover the same cell path beyond a specified overlap (extra_depth).</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.load_from_h5" title="celltraj.trajectory.Trajectory.load_from_h5"><code class="xref py py-obj docutils literal notranslate"><span class="pre">load_from_h5</span></code></a>(path)</p></td>
<td><p>Load data from a specified path within an HDF5 file.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="../celltraj.html#celltraj.trajectory.Trajectory.save_to_h5" title="celltraj.trajectory.Trajectory.save_to_h5"><code class="xref py py-obj docutils literal notranslate"><span class="pre">save_to_h5</span></code></a>(path, attribute_list[, overwrite])</p></td>
<td><p>Save specified attributes to an HDF5 file at the given path.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jeremy Copperman.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>