<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>API Reference &mdash; celltraj 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=a58bc63e"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="celltraj.trajectory" href="stubs/celltraj.trajectory.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            celltraj
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="readme.html">celltraj</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">API Reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.trajectory.html">celltraj.trajectory</a></li>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.features.html">celltraj.features</a></li>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.imageprep.html">celltraj.imageprep</a></li>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.translate.html">celltraj.translate</a></li>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.model.html">celltraj.model</a></li>
<li class="toctree-l2"><a class="reference internal" href="stubs/celltraj.utilities.html">celltraj.utilities</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.trajectory">celltraj.trajectory</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.trajectory.Trajectory"><code class="docutils literal notranslate"><span class="pre">Trajectory</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.__init__"><code class="docutils literal notranslate"><span class="pre">Trajectory.__init__()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_Xtraj_celltrajectory"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_Xtraj_celltrajectory()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_alpha"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_alpha()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_beta"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_beta()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_blocks"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_blocks()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_channel_crosscorr"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_channel_crosscorr()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_compartment_ratio"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_compartment_ratio()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_data"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_features"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_features()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_index"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_index()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.cells_frameSet"><code class="docutils literal notranslate"><span class="pre">Trajectory.cells_frameSet</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.cells_imgfileSet"><code class="docutils literal notranslate"><span class="pre">Trajectory.cells_imgfileSet</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.cells_indSet"><code class="docutils literal notranslate"><span class="pre">Trajectory.cells_indSet</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.cells_indimgSet"><code class="docutils literal notranslate"><span class="pre">Trajectory.cells_indimgSet</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.cellblocks"><code class="docutils literal notranslate"><span class="pre">Trajectory.cellblocks</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_positions"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_positions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_cell_trajectory"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_cell_trajectory()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_dx"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_dx()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_fmask_data"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_fmask_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_frames"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_frames()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_image_data"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_image_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_image_shape"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_image_shape()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.axes"><code class="docutils literal notranslate"><span class="pre">Trajectory.axes</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.nx"><code class="docutils literal notranslate"><span class="pre">Trajectory.nx</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.ny"><code class="docutils literal notranslate"><span class="pre">Trajectory.ny</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.nz"><code class="docutils literal notranslate"><span class="pre">Trajectory.nz</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.image_shape"><code class="docutils literal notranslate"><span class="pre">Trajectory.image_shape</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.nchannels"><code class="docutils literal notranslate"><span class="pre">Trajectory.nchannels</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.nmaskchannels"><code class="docutils literal notranslate"><span class="pre">Trajectory.nmaskchannels</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.ndim"><code class="docutils literal notranslate"><span class="pre">Trajectory.ndim</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_lineage_btrack"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_lineage_btrack()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_lineage_mindist"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_lineage_mindist()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_mask_data"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_mask_data()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_motility_features"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_motility_features()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_pair_rdf"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_pair_rdf()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_secreted_ligand_density"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_secreted_ligand_density()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_signal_contributions"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_signal_contributions()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_stack_trans"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_stack_trans()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_trajAB_segments"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_trajAB_segments()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_traj_segments"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_traj_segments()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_trajectory_steps"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_trajectory_steps()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.get_unique_trajectories"><code class="docutils literal notranslate"><span class="pre">Trajectory.get_unique_trajectories()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.load_from_h5"><code class="docutils literal notranslate"><span class="pre">Trajectory.load_from_h5()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#celltraj.trajectory.Trajectory.save_to_h5"><code class="docutils literal notranslate"><span class="pre">Trajectory.save_to_h5()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.features">celltraj.features</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.apply3d"><code class="docutils literal notranslate"><span class="pre">apply3d()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.boundaryCB_FFT"><code class="docutils literal notranslate"><span class="pre">boundaryCB_FFT()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.boundaryFFT"><code class="docutils literal notranslate"><span class="pre">boundaryFFT()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featBoundary"><code class="docutils literal notranslate"><span class="pre">featBoundary()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featBoundaryCB"><code class="docutils literal notranslate"><span class="pre">featBoundaryCB()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featHaralick"><code class="docutils literal notranslate"><span class="pre">featHaralick()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featNucBoundary"><code class="docutils literal notranslate"><span class="pre">featNucBoundary()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featSize"><code class="docutils literal notranslate"><span class="pre">featSize()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.featZernike"><code class="docutils literal notranslate"><span class="pre">featZernike()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.get_cc_cs_border"><code class="docutils literal notranslate"><span class="pre">get_cc_cs_border()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.get_contact_boundaries"><code class="docutils literal notranslate"><span class="pre">get_contact_boundaries()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.get_contact_labels"><code class="docutils literal notranslate"><span class="pre">get_contact_labels()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.get_neighbor_feature_map"><code class="docutils literal notranslate"><span class="pre">get_neighbor_feature_map()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.get_pca_fromdata"><code class="docutils literal notranslate"><span class="pre">get_pca_fromdata()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.meanIntensity"><code class="docutils literal notranslate"><span class="pre">meanIntensity()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.features.totalIntensity"><code class="docutils literal notranslate"><span class="pre">totalIntensity()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.imageprep">celltraj.imageprep</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.clean_labeled_mask"><code class="docutils literal notranslate"><span class="pre">clean_labeled_mask()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.create_h5"><code class="docutils literal notranslate"><span class="pre">create_h5()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.crop_image"><code class="docutils literal notranslate"><span class="pre">crop_image()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.dist_to_contact"><code class="docutils literal notranslate"><span class="pre">dist_to_contact()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.expand_registered_images"><code class="docutils literal notranslate"><span class="pre">expand_registered_images()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_cell_centers"><code class="docutils literal notranslate"><span class="pre">get_cell_centers()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_cell_intensities"><code class="docutils literal notranslate"><span class="pre">get_cell_intensities()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_contactsum_dev"><code class="docutils literal notranslate"><span class="pre">get_contactsum_dev()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_cyto_minus_nuc_labels"><code class="docutils literal notranslate"><span class="pre">get_cyto_minus_nuc_labels()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_feature_map"><code class="docutils literal notranslate"><span class="pre">get_feature_map()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_images"><code class="docutils literal notranslate"><span class="pre">get_images()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_intensity_centers"><code class="docutils literal notranslate"><span class="pre">get_intensity_centers()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_label_largestcc"><code class="docutils literal notranslate"><span class="pre">get_label_largestcc()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_labeled_mask"><code class="docutils literal notranslate"><span class="pre">get_labeled_mask()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_mask_2channel_ilastik"><code class="docutils literal notranslate"><span class="pre">get_mask_2channel_ilastik()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_masks"><code class="docutils literal notranslate"><span class="pre">get_masks()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_nndist_sum"><code class="docutils literal notranslate"><span class="pre">get_nndist_sum()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_pair_rdf_fromcenters"><code class="docutils literal notranslate"><span class="pre">get_pair_rdf_fromcenters()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_registration_expansions"><code class="docutils literal notranslate"><span class="pre">get_registration_expansions()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_registrations"><code class="docutils literal notranslate"><span class="pre">get_registrations()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_slide_image"><code class="docutils literal notranslate"><span class="pre">get_slide_image()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_tf_matrix_2d"><code class="docutils literal notranslate"><span class="pre">get_tf_matrix_2d()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_tile_order"><code class="docutils literal notranslate"><span class="pre">get_tile_order()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_voronoi_masks"><code class="docutils literal notranslate"><span class="pre">get_voronoi_masks()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.get_voronoi_masks_fromcenters"><code class="docutils literal notranslate"><span class="pre">get_voronoi_masks_fromcenters()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.histogram_stretch"><code class="docutils literal notranslate"><span class="pre">histogram_stretch()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.list_images"><code class="docutils literal notranslate"><span class="pre">list_images()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.load_for_viewing"><code class="docutils literal notranslate"><span class="pre">load_for_viewing()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.load_ilastik"><code class="docutils literal notranslate"><span class="pre">load_ilastik()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.local_threshold"><code class="docutils literal notranslate"><span class="pre">local_threshold()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.make_odd"><code class="docutils literal notranslate"><span class="pre">make_odd()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.organize_filelist_fov"><code class="docutils literal notranslate"><span class="pre">organize_filelist_fov()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.organize_filelist_time"><code class="docutils literal notranslate"><span class="pre">organize_filelist_time()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.pad_image"><code class="docutils literal notranslate"><span class="pre">pad_image()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.save_for_viewing"><code class="docutils literal notranslate"><span class="pre">save_for_viewing()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.save_frame_h5"><code class="docutils literal notranslate"><span class="pre">save_frame_h5()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.transform_image"><code class="docutils literal notranslate"><span class="pre">transform_image()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.imageprep.znorm"><code class="docutils literal notranslate"><span class="pre">znorm()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.model">celltraj.model</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.clean_clusters"><code class="docutils literal notranslate"><span class="pre">clean_clusters()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_H_eigs"><code class="docutils literal notranslate"><span class="pre">get_H_eigs()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_avdx_clusters"><code class="docutils literal notranslate"><span class="pre">get_avdx_clusters()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_committor"><code class="docutils literal notranslate"><span class="pre">get_committor()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_dmat"><code class="docutils literal notranslate"><span class="pre">get_dmat()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_gaussianKernelM"><code class="docutils literal notranslate"><span class="pre">get_gaussianKernelM()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_kernel_sigmas"><code class="docutils literal notranslate"><span class="pre">get_kernel_sigmas()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_kineticstates"><code class="docutils literal notranslate"><span class="pre">get_kineticstates()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_koopman_eig"><code class="docutils literal notranslate"><span class="pre">get_koopman_eig()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_koopman_inference"><code class="docutils literal notranslate"><span class="pre">get_koopman_inference()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_koopman_inference_multiple"><code class="docutils literal notranslate"><span class="pre">get_koopman_inference_multiple()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_koopman_modes"><code class="docutils literal notranslate"><span class="pre">get_koopman_modes()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_kscore"><code class="docutils literal notranslate"><span class="pre">get_kscore()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_landscape_coords_umap"><code class="docutils literal notranslate"><span class="pre">get_landscape_coords_umap()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_motifs"><code class="docutils literal notranslate"><span class="pre">get_motifs()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_path_entropy_2point"><code class="docutils literal notranslate"><span class="pre">get_path_entropy_2point()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_path_ll_2point"><code class="docutils literal notranslate"><span class="pre">get_path_ll_2point()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_steady_state_matrixpowers"><code class="docutils literal notranslate"><span class="pre">get_steady_state_matrixpowers()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_traj_ll_gmean"><code class="docutils literal notranslate"><span class="pre">get_traj_ll_gmean()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_transition_matrix"><code class="docutils literal notranslate"><span class="pre">get_transition_matrix()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.get_transition_matrix_CG"><code class="docutils literal notranslate"><span class="pre">get_transition_matrix_CG()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.plot_dx_arrows"><code class="docutils literal notranslate"><span class="pre">plot_dx_arrows()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.plot_eig"><code class="docutils literal notranslate"><span class="pre">plot_eig()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.update_mahalanobis_matrix_J"><code class="docutils literal notranslate"><span class="pre">update_mahalanobis_matrix_J()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.update_mahalanobis_matrix_J_old"><code class="docutils literal notranslate"><span class="pre">update_mahalanobis_matrix_J_old()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.update_mahalanobis_matrix_flux"><code class="docutils literal notranslate"><span class="pre">update_mahalanobis_matrix_flux()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.model.update_mahalanobis_matrix_grad"><code class="docutils literal notranslate"><span class="pre">update_mahalanobis_matrix_grad()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.translate">celltraj.translate</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.translate.get_null_correlations"><code class="docutils literal notranslate"><span class="pre">get_null_correlations()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.translate.get_predictedFC"><code class="docutils literal notranslate"><span class="pre">get_predictedFC()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.translate.get_state_decomposition"><code class="docutils literal notranslate"><span class="pre">get_state_decomposition()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-celltraj.utilities">celltraj.utilities</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.colorbar"><code class="docutils literal notranslate"><span class="pre">colorbar()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.dist"><code class="docutils literal notranslate"><span class="pre">dist()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.dist_to_contact"><code class="docutils literal notranslate"><span class="pre">dist_to_contact()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_cdist"><code class="docutils literal notranslate"><span class="pre">get_cdist()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_cdist2d"><code class="docutils literal notranslate"><span class="pre">get_cdist2d()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_cell_centers"><code class="docutils literal notranslate"><span class="pre">get_cell_centers()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_dmat"><code class="docutils literal notranslate"><span class="pre">get_dmat()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_dmat_vectorized"><code class="docutils literal notranslate"><span class="pre">get_dmat_vectorized()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_linear_batch_normalization"><code class="docutils literal notranslate"><span class="pre">get_linear_batch_normalization()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_linear_coef"><code class="docutils literal notranslate"><span class="pre">get_linear_coef()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_meshfunc_average"><code class="docutils literal notranslate"><span class="pre">get_meshfunc_average()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_pairwise_distance_sum"><code class="docutils literal notranslate"><span class="pre">get_pairwise_distance_sum()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.get_tshift"><code class="docutils literal notranslate"><span class="pre">get_tshift()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.load_dict_from_h5"><code class="docutils literal notranslate"><span class="pre">load_dict_from_h5()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.recursively_load_dict_contents_from_group"><code class="docutils literal notranslate"><span class="pre">recursively_load_dict_contents_from_group()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.recursively_save_dict_contents_to_group"><code class="docutils literal notranslate"><span class="pre">recursively_save_dict_contents_to_group()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#celltraj.utilities.save_dict_to_h5"><code class="docutils literal notranslate"><span class="pre">save_dict_to_h5()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">celltraj</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">API Reference</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/api.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="api-reference">
<h1>API Reference<a class="headerlink" href="#api-reference" title="Link to this heading"></a></h1>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.trajectory" title="celltraj.trajectory"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.trajectory</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.features" title="celltraj.features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.features</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.imageprep" title="celltraj.imageprep"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.imageprep</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.translate" title="celltraj.translate"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.translate</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.model" title="celltraj.model"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.model</span></code></a></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="celltraj.html#module-celltraj.utilities" title="celltraj.utilities"><code class="xref py py-obj docutils literal notranslate"><span class="pre">celltraj.utilities</span></code></a></p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<section id="module-celltraj.trajectory">
<span id="celltraj-trajectory"></span><h2>celltraj.trajectory<a class="headerlink" href="#module-celltraj.trajectory" title="Link to this heading"></a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">celltraj.trajectory.</span></span><span class="sig-name descname"><span class="pre">Trajectory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5filename</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A toolset for single-cell trajectory modeling. See:</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This code, currently, should be considered as an untested pre-release version</p>
</div>
<p class="rubric">References</p>
<p>Copperman, Jeremy, Sean M. Gross, Young Hwan Chang, Laura M. Heiser, and Daniel M. Zuckerman.
“Morphodynamical cell state description via live-cell imaging trajectory embedding.”
Communications Biology 6, no. 1 (2023): 484.</p>
<p>Copperman, Jeremy, Ian C. Mclean, Sean M. Gross, Young Hwan Chang, Daniel M. Zuckerman, and Laura M. Heiser.
“Single-cell morphodynamical trajectories enable prediction of gene expression accompanying cell state change.”
bioRxiv (2024): 2024-01.</p>
<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5filename</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.__init__"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a Trajectory object, optionally loading metadata from an HDF5 file.</p>
<p>This constructor sets the HDF5 filename and attempts to load metadata associated with the file.
If the file is present, it reads the metadata from a predefined group. Errors during metadata
loading are caught and logged. Future updates should include better commenting and organizational
improvements of class attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>h5filename</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – The path to the HDF5 file from which to load the metadata. If not provided, the
instance will be initialized without loading metadata.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>TODO:
- Improve documentation of class attributes.
- Reorganize attributes into a more meaningful structure.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="go">loading path/to/your/hdf5file.h5</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_Xtraj_celltrajectory">
<span class="sig-name descname"><span class="pre">get_Xtraj_celltrajectory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cell_traj</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Xtraj</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">traj</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_Xtraj_celltrajectory"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_Xtraj_celltrajectory" title="Link to this definition"></a></dt>
<dd><p>Retrieves trajectory segments for a specific cell trajectory from a larger set of trajectory data.
This method matches segments of the cell trajectory with those in a pre-computed set of trajectories
and extracts the corresponding features or data points.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cell_traj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – An array containing indices of a cell’s trajectory over time.</p></li>
<li><p><strong>Xtraj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – The trajectory feature matrix from which to extract data. If not provided, the method uses
the instance’s attribute <cite>Xtraj</cite>.</p></li>
<li><p><strong>traj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – A matrix of precomputed trajectories used for matching against <cite>cell_traj</cite>. If not provided,
the method uses the instance’s attribute <cite>traj</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>xt</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – A subset of <cite>Xtraj</cite> corresponding to the segments of <cite>cell_traj</cite> that match segments in <cite>traj</cite>.</p></li>
<li><p><strong>inds_traj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Indices within <cite>traj</cite> where matches were found, indicating which rows in <cite>Xtraj</cite> were selected.</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If the length of <cite>cell_traj</cite> is less than the length used for trajectories in <cite>traj</cite> (<cite>trajl</cite>),
    making it impossible to match any trajectory segments.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">get_unique_trajectories</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cell_trajectory</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_trajectory</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">features</span><span class="p">,</span> <span class="n">indices</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_Xtraj_celltrajectory</span><span class="p">(</span><span class="n">cell_trajectory</span><span class="p">)</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The method requires <cite>trajl</cite>, the length of the trajectory segments, to be set either as a class</p></li>
</ul>
<p>attribute or passed explicitly. This length determines how the segments are compared for matching.
- This function is particularly useful for analyzing time-series data or features extracted from
trajectories, allowing for detailed analysis specific to a single cell’s path through time.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_alpha">
<span class="sig-name descname"><span class="pre">get_alpha</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">i1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_alpha"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_alpha" title="Link to this definition"></a></dt>
<dd><p>Calculates the alignment measure, alpha, between two cells identified by their indices. This
measure reflects how the movement direction of one cell relates to the direction of the vector
connecting the two cells, essentially quantifying the relative motion along the axis of separation.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>i1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Index of the first cell.</p></li>
<li><p><strong>i2</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Index of the second cell.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>alpha</strong> – The alignment measure between the two cells. This value ranges from -1 to 1, where 1 indicates
that the cells are moving directly towards each other, -1 indicates they are moving directly
away from each other, and 0 indicates orthogonal movement directions. Returns NaN if the calculation
fails (e.g., due to division by zero when normalizing zero-length vectors).</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>Exception</strong> – If an error occurs during the trajectory retrieval or normalization process, likely due to missing
    data or incorrect indices.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">alignment</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_alpha</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">)</span>
<span class="go"># This computes the alignment measure between cells at index 10 and 15 based on their last movements.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The function computes the movement vectors of both cells from their previous positions in their</p></li>
</ul>
<p>respective trajectories and uses these vectors to determine their alignment relative to the vector
connecting the two cells at their current positions.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_beta">
<span class="sig-name descname"><span class="pre">get_beta</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">i1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_beta"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_beta" title="Link to this definition"></a></dt>
<dd><p>Calculates the cosine of the angle (beta) between the movement directions of two cells. This measure
quantifies the directional similarity or alignment between two moving cells, with values ranging
from -1 to 1.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>i1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Index of the first cell.</p></li>
<li><p><strong>i2</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Index of the second cell.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>beta</strong> – The cosine of the angle between the movement vectors of the two cells, indicating their
directional alignment. A value of 1 means the cells are moving in exactly the same direction,
-1 means they are moving in exactly opposite directions, and 0 indicates orthogonal movement
directions. Returns NaN if the calculation fails, typically due to a division by zero when
attempting to normalize zero-length vectors.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>Exception</strong> – If an error occurs during the trajectory retrieval or normalization process, likely due to missing
    data or incorrect indices.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">alignment</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_beta</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">)</span>
<span class="go"># This computes the directional alignment between cells at index 10 and 15 based on their last movements.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The function calculates movement vectors for both cells from their positions at the last two time points</p></li>
</ul>
<p>in their trajectories. It then computes the cosine of the angle between these vectors as the beta value,
providing an indication of how parallel their movements are.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_blocks">
<span class="sig-name descname"><span class="pre">get_cell_blocks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">label</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_blocks"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_blocks" title="Link to this definition"></a></dt>
<dd><p>Extracts bounding box information for each cell from a labeled mask image. This function
returns the minimum and maximum indices for each labeled cell, useful for operations such
as cropping around a cell or analyzing specific cell regions. The function supports both
2D and 3D labeled images.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>label</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – A labeled image array where each unique non-zero integer represents a unique cell.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>cellblocks</strong> – An array containing the bounding boxes for each cell. The array has shape
(number_of_labels, number_of_dimensions, 2), where each cell’s bounding box is
represented by [min_dim1, min_dim2, …, max_dim1, max_dim2, …].</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">label_image</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">blocks</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_blocks</span><span class="p">(</span><span class="n">label_image</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">blocks</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(2, 2, 2)  # Example output shape for a 2D label image with two labels.</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_channel_crosscorr">
<span class="sig-name descname"><span class="pre">get_cell_channel_crosscorr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">indcells</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgchannel1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgchannel2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_channel_crosscorr"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_channel_crosscorr" title="Link to this definition"></a></dt>
<dd><p>Computes the cross-correlation between two image channels within labeled cells, using masks defined by a specific mask channel. This method is particularly useful for analyzing the relationship between different signal channels at a cellular level.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>indcells</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of cells for which to calculate cross-correlations. If None, calculations are performed for all indexed cells.</p></li>
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Mask channel used to define cellular regions.</p></li>
<li><p><strong>imgchannel1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – First image channel for correlation analysis.</p></li>
<li><p><strong>imgchannel2</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Second image channel for correlation analysis.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the calculated cross-correlations to an HDF5 file under the specified directory and file names.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is True, existing data in the HDF5 file will be overwritten.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>corrc</strong> – Array of cross-correlation coefficients for each cell. The length of the array corresponds to the number of cells specified by <cite>indcells</cite>.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If required parameters are not set or if no cell index is available, prompting the user to set necessary parameters or perform required prior steps.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">corrc</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_channel_crosscorr</span><span class="p">(</span><span class="n">indcells</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="n">mskchannel</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">imgchannel1</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">imgchannel2</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_compartment_ratio">
<span class="sig-name descname"><span class="pre">get_cell_compartment_ratio</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">indcells</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmask_channel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">make_disjoint</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">erosion_footprint1</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">erosion_footprint2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">combined_and_disjoint</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity_sum</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity_ztransform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">noratio</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse_ratio</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_compartment_ratio"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_compartment_ratio" title="Link to this definition"></a></dt>
<dd><p>Calculates the ratio of features between two cellular compartments, optionally adjusted for image
intensity and morphology transformations. This method allows for complex comparisons between different
mask channels or modified versions of these channels to derive cellular compartment ratios.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>indcells</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of cells for which to calculate the feature ratios.</p></li>
<li><p><strong>imgchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Index of the image channel used for intensity measurements.</p></li>
<li><p><strong>mskchannel1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Mask channel index for the numerator in the ratio calculation.</p></li>
<li><p><strong>mskchannel2</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Mask channel index for the denominator in the ratio calculation. Overlaps with mskchannel1 are removed.</p></li>
<li><p><strong>fmask_channel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Mask channel index used to adjust mskchannel2 if no separate mskchannel2 is provided.</p></li>
<li><p><strong>make_disjoint</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, ensures that masks from mskchannel1 and mskchannel2 do not overlap by adjusting mskchannel1.</p></li>
<li><p><strong>erosion_footprint1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – Erosion footprint for the first mask, modifies the mask by eroding it before calculations.</p></li>
<li><p><strong>erosion_footprint2</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – Erosion footprint for the second mask, modifies the mask by eroding it before calculations.</p></li>
<li><p><strong>combined_and_disjoint</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, combines and then separates the masks to only include disjoint areas in calculations.</p></li>
<li><p><strong>intensity_sum</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, sums the intensity over the area rather than averaging it, before ratio calculation.</p></li>
<li><p><strong>intensity_ztransform</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, applies a z-transformation based on standard deviations and means stored in the object.</p></li>
<li><p><strong>noratio</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, returns only the numerator intensity mean without forming a ratio.</p></li>
<li><p><strong>inverse_ratio</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, calculates the inverse of the normal ratio.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the calculated ratios to an HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is also True, overwrites existing data in the HDF5 file.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>cratio</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Array of calculated compartment ratios for each specified cell.</p></li>
<li><p><strong>feature_list</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – Descriptions of the cell features, returned if <cite>return_feature_list</cite> is set to True.</p></li>
</ul>
</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cratio</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_compartment_ratio</span><span class="p">(</span><span class="n">indcells</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span> <span class="n">imgchannel</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">mskchannel1</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">mskchannel2</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_data">
<span class="sig-name descname"><span class="pre">get_cell_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ic</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">frametype</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'boundingbox'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary_expansion</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_masks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relabel_masks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relabel_mskchannels</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">delete_background</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_data" title="Link to this definition"></a></dt>
<dd><p>Retrieves image and mask data for specific cells based on various configuration options.
This method can extract data for single cells, their neighborhoods, or connected cell groups,
and offers options to expand the extraction region, relabel masks, and more.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ic</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Cell ID(s) for which to retrieve data. Can specify a single cell or a list of cells.</p></li>
<li><p><strong>frametype</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – Type of frame data to retrieve; options include ‘boundingbox’, ‘neighborhood’, or ‘connected’.
Default is ‘boundingbox’.</p></li>
<li><p><strong>boundary_expansion</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Array specifying how much to expand the bounding box around the cell in each dimension.</p></li>
<li><p><strong>return_masks</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – Whether to return the mask data along with the image data. Default is True.</p></li>
<li><p><strong>relabel_masks</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – Whether to relabel mask data with movie cell indices. Default is True.</p></li>
<li><p><strong>relabel_mskchannels</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code>, <em>optional</em>) – Specifies the mask channels to relabel. If not set, uses the default mask channel.</p></li>
<li><p><strong>delete_background</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If set to True, sets label and image pixels outside the specified cell set to zero. Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>imgc</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – The image data for the specified cell(s) as a NumPy array.</p></li>
<li><p><strong>mskc</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – The mask data for the specified cell(s) as a NumPy array, returned if <cite>return_masks</cite> is True.</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If cells from multiple frames are requested or necessary attributes are not set.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">img_data</span><span class="p">,</span> <span class="n">mask_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_data</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="n">frametype</span><span class="o">=</span><span class="s1">&#39;boundingbox&#39;</span><span class="p">,</span> <span class="n">boundary_expansion</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">return_masks</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_features">
<span class="sig-name descname"><span class="pre">get_cell_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">function_tuple</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">indcells</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_fmask_for_intensity_image</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_mask_for_intensity_image</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bordersize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apply_contact_transform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_feature_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">concatenate_features</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_features"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_features" title="Link to this definition"></a></dt>
<dd><p>Extracts complex cell features based on the specified custom functions and imaging data.
This method allows customization of the feature extraction process using region properties
from segmented cell data, with optional transformations like border erosion or contact
transformations. Features can be based on intensity images, mask data, or specific
transformations of these data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>function_tuple</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">callable</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">tuple</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">callables</span></code>) – Function(s) that take a mask and an intensity image as input and return a scalar
or array of features. These functions must be compatible with skimage’s regionprops.</p></li>
<li><p><strong>indcells</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Array of cell indices for which to calculate features. If None, calculates for all cells.</p></li>
<li><p><strong>imgchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Index of the image channel used for intensity image feature calculation.</p></li>
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Index of the mask channel used for single-cell label feature calculation.</p></li>
<li><p><strong>use_fmask_for_intensity_image</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, uses foreground mask data for intensity images.</p></li>
<li><p><strong>fmskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Channel index for foreground mask data if used for intensity image.</p></li>
<li><p><strong>use_mask_for_intensity_image</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, uses mask data instead of image data for intensity measurements.</p></li>
<li><p><strong>bordersize</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Pixel size for erosion of the foreground mask or growth radius for contact boundaries.</p></li>
<li><p><strong>apply_contact_transform</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, applies a contact transform to generate segmentation contacts from the mask data.</p></li>
<li><p><strong>return_feature_list</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, returns a list of strings describing the calculated features.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the features and their descriptions to an HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is also True, overwrites existing data in the HDF5 file.</p></li>
<li><p><strong>concatenate_features</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, adds the newly calculated features to existing features in the dataset.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>Xf</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Array of features indexed by cells. The shape is (number of cells, number of features).</p></li>
<li><p><strong>feature_list</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – Array of strings describing each feature, returned if <cite>return_feature_list</cite> is True.</p></li>
</ul>
</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">features</span><span class="p">,</span> <span class="n">feature_descriptions</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_features</span><span class="p">(</span><span class="n">my_feature_funcs</span><span class="p">,</span> <span class="n">return_feature_list</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_index">
<span class="sig-name descname"><span class="pre">get_cell_index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_index"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_index" title="Link to this definition"></a></dt>
<dd><p>Computes indices and corresponding frame information for each cell in an image stack, capturing
this data in several attributes. This method gathers extensive cell data across all frames,
including frame indices, image file indices, individual image indices, and bounding boxes
for each cell. This information is stored in corresponding attributes, facilitating further
analysis or reference.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>verbose</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, prints detailed logging of the processing for each frame. Default is False.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the computed data to an HDF5 file using the specified <cite>mskchannel</cite>. Default is False.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and <cite>save_h5</cite> is True, existing data in the HDF5 file will be overwritten. Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>True if the computation and any specified data saving are successful, False if there is
an error due to missing prerequisites or during saving.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></p>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.cells_frameSet">
<span class="sig-name descname"><span class="pre">cells_frameSet</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.cells_frameSet" title="Link to this definition"></a></dt>
<dd><p>Array storing the frame index for each cell, shape (ncells_total,).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.cells_imgfileSet">
<span class="sig-name descname"><span class="pre">cells_imgfileSet</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.cells_imgfileSet" title="Link to this definition"></a></dt>
<dd><p>Array storing the image file index for each cell, shape (ncells_total,).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.cells_indSet">
<span class="sig-name descname"><span class="pre">cells_indSet</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.cells_indSet" title="Link to this definition"></a></dt>
<dd><p>Array storing a unique index for each cell in the trajectory, shape (ncells_total,).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.cells_indimgSet">
<span class="sig-name descname"><span class="pre">cells_indimgSet</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.cells_indimgSet" title="Link to this definition"></a></dt>
<dd><p>Array storing the image-specific index for each cell, shape (ncells_total,).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.cellblocks">
<span class="sig-name descname"><span class="pre">cellblocks</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.cellblocks" title="Link to this definition"></a></dt>
<dd><p>Array of bounding boxes for each cell, shape (ncells_total, ndim, 2).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>AttributeError</strong> – If necessary attributes (like <cite>nmaskchannels</cite>, <cite>ncells_total</cite>, <cite>mskchannel</cite>, <cite>maxFrame</cite>)
    are not set prior to invoking this method.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">success</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_index</span><span class="p">(</span><span class="n">verbose</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">save_h5</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">overwrite</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_positions">
<span class="sig-name descname"><span class="pre">get_cell_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_positions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_positions" title="Link to this definition"></a></dt>
<dd><p>Calculate the center of mass for cells in each frame of the mask channel and optionally save
these positions to an HDF5 file. This method processes mask data to find cell positions across
frames and can store these positions back into the HDF5 file associated with the Trajectory
instance.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The index of the mask channel from which to calculate cell positions. Default is 0.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, the calculated cell positions will be saved to the HDF5 file specified by
<cite>h5filename</cite>. Default is False.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and <cite>save_h5</cite> is also True, existing data in the HDF5 file will be overwritten.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>An array of cell positions calculated from the mask channel. The shape of the array is
(number of cells, number of dimensions).</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>RuntimeError</strong> – If the stack has not been transformed, indicated by <cite>tf_matrix_set</cite> not being set.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">get_stack_trans</span><span class="p">()</span>  <span class="c1"># Ensure transformation matrix is set</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">positions</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_positions</span><span class="p">(</span><span class="n">mskchannel</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">save_h5</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">overwrite</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">getting positions from mask channel 1, default mskchannel is 0</span>
<span class="go">loading cells from frame 0</span>
<span class="go">loading cells from frame 1</span>
<span class="go">...</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_cell_trajectory">
<span class="sig-name descname"><span class="pre">get_cell_trajectory</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cell_ind</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_hist</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_cell_trajectory"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_cell_trajectory" title="Link to this definition"></a></dt>
<dd><p>Retrieves the trajectory of a specified cell across previous frames, tracing back from the current frame to
the point of its first appearance or until a specified number of history steps.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cell_ind</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The index of the cell for which to retrieve the trajectory.</p></li>
<li><p><strong>n_hist</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The number of historical steps to trace back. If set to -1 (default), the function traces back
to the earliest frame in which the cell appears.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>cell_traj</strong> – An array of cell indices representing the trajectory of the specified cell across the tracked frames.
The array is ordered from the earliest appearance to the current frame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>IndexError</strong> – If the cell index provided is out of the bounds of the available data.</p></li>
<li><p><strong>ValueError</strong> – If the provided cell index does not correspond to any tracked cell, possibly due to errors in lineage tracking.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cell_trajectory</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_cell_trajectory</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cell_trajectory</span><span class="p">)</span>
<span class="go">[23, 45, 67, 89]  # Example output, actual values depend on cell tracking results.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<p>The trajectory is computed by accessing the lineage data (<cite>linSet</cite>), which must be computed beforehand
via methods such as <cite>get_lineage_btrack</cite>. Each index in the resulting trajectory corresponds to a
position in previous frames where the cell was identified, stepping backwards until the cell’s first
detection or the limit of specified history steps.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_dx">
<span class="sig-name descname"><span class="pre">get_dx</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">i1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_dx"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_dx" title="Link to this definition"></a></dt>
<dd><p>Calculates the displacement vector of a cell between its current position and its previous position in the
trajectory. This vector represents the movement of the cell between two consecutive time points.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>i1</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Index of the cell for which to calculate the displacement.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>dx1</strong> – A vector representing the displacement of the cell. The vector is given in the coordinate space
of the cell positions. If the calculation fails (e.g., due to missing data), returns a vector
of NaNs.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>Exception</strong> – If an error occurs during the trajectory retrieval or calculation, typically due to missing
    data or incorrect indices.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">displacement</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_dx</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
<span class="go"># This calculates the displacement vector for the cell at index 10 between its current and previous positions.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The method attempts to retrieve the last position from the cell’s trajectory using <cite>get_cell_trajectory</cite>.</p></li>
</ul>
<p>If the cell’s trajectory does not have a previous position or the data is missing, the displacement
vector will contain NaN values to indicate the failure of the calculation.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_fmask_data">
<span class="sig-name descname"><span class="pre">get_fmask_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_frame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">channel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_fmask_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_fmask_data" title="Link to this definition"></a></dt>
<dd><p>Retrieve the foreground mask data for a specific frame using different methods depending on
the set attributes. This method determines the foreground mask either by selecting a specific
mask channel (<cite>fmskchannel</cite>), by applying a threshold to an image channel (<cite>fmsk_threshold</cite>
and <cite>fmsk_imgchannel</cite>), or directly from specified channels in the HDF5 file (<cite>fmask_channels</cite>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n_frame</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The frame number from which to retrieve the foreground mask data.</p></li>
<li><p><strong>channel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The specific channel to use when <cite>fmask_channels</cite> attribute is set. If not provided, the
default ‘foreground’ channel is used if available.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>fmsk</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – The foreground (cells) / background mask array, indicating cell locations as True and
background as False.</p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">Methods</span> <span class="pre">for</span> <span class="pre">Determining</span> <span class="pre">`fmsk`</span></code></p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">------------------------------</span></code></p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">1.</span> <span class="pre">If</span> <span class="pre">`fmskchannel</span></code> is set`</p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">-</span> <span class="pre">The</span> <span class="pre">method</span> <span class="pre">uses</span> <span class="pre">the</span> <span class="pre">specified</span> <span class="pre">channel</span> <span class="pre">from</span> <span class="pre">the</span> <span class="pre">mask</span> <span class="pre">data</span> <span class="pre">(not</span> <span class="pre">fmask</span> <span class="pre">data).</span></code></p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">2.</span> <span class="pre">If</span> <span class="pre">`fmsk_threshold`</span></code> and <cite>fmsk_imgchannel</cite> are set</p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">-</span> <span class="pre">The</span> <span class="pre">method</span> <span class="pre">thresholds</span> <span class="pre">the</span> <span class="pre">image</span> <span class="pre">data</span> <span class="pre">at</span> <span class="pre">the</span> <span class="pre">specified</span> <span class="pre">channel</span> <span class="pre">using</span> <span class="pre">the</span> <span class="pre">given</span> <span class="pre">threshold.</span></code></p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">3.</span> <span class="pre">If</span> <span class="pre">`fmask_channels</span></code> is set` and <code class="xref py py-class docutils literal notranslate"><span class="pre">the</span> <span class="pre">channel</span> <span class="pre">parameter</span> <span class="pre">is</span> <span class="pre">provided</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">a</span> <span class="pre">default</span> <span class="pre">is</span> <span class="pre">available</span></code></p></li>
<li><p><code class="xref py py-class docutils literal notranslate"><span class="pre">-</span> <span class="pre">The</span> <span class="pre">method</span> <span class="pre">retrieves</span> <span class="pre">the</span> <span class="pre">mask</span> <span class="pre">from</span> <span class="pre">the</span> <span class="pre">specified</span></code> or <em>default</em> <code class="xref py py-class docutils literal notranslate"><span class="pre">channel</span> <span class="pre">in</span> <span class="pre">the</span> <span class="pre">`fmsk</span></code> dataset.`</p></li>
</ul>
</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fmask_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_fmask_data</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fmask_data</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1024, 1024)  # Example shape, actual may vary.</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_frames">
<span class="sig-name descname"><span class="pre">get_frames</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_frames"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_frames" title="Link to this definition"></a></dt>
<dd><p>Scans an HDF5 file for image and mask datasets to determine the total number of frames,
images per frame, and the total number of cells across all frames. This method updates the
instance with attributes for the number of images, the maximum frame index, and the total
cell count. It handles multiple mask channels by requiring the <cite>mskchannel</cite> attribute to be
set if more than one mask channel exists.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>Returns True if the frames were successfully scanned and the relevant attributes set.
Returns False if the HDF5 file is not set, no image data is found, or if there are
multiple mask channels but <cite>mskchannel</cite> is not specified.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>AttributeError</strong> – If <cite>mskchannel</cite> needs to be specified but is not set when <cite>nmaskchannel</cite> is greater than zero.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">success</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_frames</span><span class="p">()</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_image_data">
<span class="sig-name descname"><span class="pre">get_image_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_frame</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_image_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_image_data" title="Link to this definition"></a></dt>
<dd><p>Retrieve the image data for a specified frame from the HDF5 file associated with this instance.
This method accesses the HDF5 file, navigates to the specific dataset for the given frame,
and extracts the image data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>n_frame</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The frame number from which to retrieve image data.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>img</strong> – The image data as a NumPy array. The shape and type of the array depend on the structure
of the image data in the HDF5 file (e.g., may include dimensions for channels or z-stacks).</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">image_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_image_data</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">image_data</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1024, 1024, 3)  # Example shape, actual may vary.</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_image_shape">
<span class="sig-name descname"><span class="pre">get_image_shape</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_frame</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_image_shape"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_image_shape" title="Link to this definition"></a></dt>
<dd><p>Determine the dimensions of the image and mask data from the HDF5 file at a specified frame
index and store these as attributes. This method retrieves the dimensions of both the image
and mask datasets, discerning whether the data includes channels or z-stacks, and updates the
object’s attributes accordingly. Attributes updated include the number of dimensions (<cite>ndim</cite>),
image axes layout (<cite>axes</cite>), image dimensions (<cite>nx</cite>, <cite>ny</cite>, <cite>[nz]</cite>), number of channels in the
image and mask (<cite>nchannels</cite>, <cite>nmaskchannels</cite>), and the full image shape (<cite>image_shape</cite>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>n_frame</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The frame index from which to retrieve the image and mask dimensions. Default is 0.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Returns True if the dimensions were successfully retrieved and stored as attributes,
False otherwise, such as when the file is not found or an error occurs in reading data.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></p>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.axes">
<span class="sig-name descname"><span class="pre">axes</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.axes" title="Link to this definition"></a></dt>
<dd><p>The layout of axes in the image data, e.g., ‘xy’, ‘xyc’, ‘zxy’, ‘zxyc’.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.nx">
<span class="sig-name descname"><span class="pre">nx</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.nx" title="Link to this definition"></a></dt>
<dd><p>Width of the image in pixels.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.ny">
<span class="sig-name descname"><span class="pre">ny</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.ny" title="Link to this definition"></a></dt>
<dd><p>Height of the image in pixels.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.nz">
<span class="sig-name descname"><span class="pre">nz</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.nz" title="Link to this definition"></a></dt>
<dd><p>Number of z-stacks in the image, if applicable.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.image_shape">
<span class="sig-name descname"><span class="pre">image_shape</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.image_shape" title="Link to this definition"></a></dt>
<dd><p>Array representing the dimensions of the image.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.nchannels">
<span class="sig-name descname"><span class="pre">nchannels</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.nchannels" title="Link to this definition"></a></dt>
<dd><p>Number of channels in the image data.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.nmaskchannels">
<span class="sig-name descname"><span class="pre">nmaskchannels</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.nmaskchannels" title="Link to this definition"></a></dt>
<dd><p>Number of channels in the mask data.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.ndim">
<span class="sig-name descname"><span class="pre">ndim</span></span><a class="headerlink" href="#celltraj.trajectory.Trajectory.ndim" title="Link to this definition"></a></dt>
<dd><p>Number of spatial dimensions in the image data.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code></p>
</dd>
</dl>
</dd></dl>

<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">success</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_image_shape</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_lineage_btrack">
<span class="sig-name descname"><span class="pre">get_lineage_btrack</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">framewindow</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visual_1cell</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visual</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_search_radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_lineage_btrack"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_lineage_btrack" title="Link to this definition"></a></dt>
<dd><p>Tracks cell lineages over an image stack using Bayesian tracking with visual confirmation options.
This method registers transformed masks and applies Bayesian tracking to link cell identities
across frames, storing the lineage information.
Use of btrack software requires a cell_config.json file stored in the directory, see btrack documentation.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Mask channel used to identify cell labels from which cell centers are extracted.</p></li>
<li><p><strong>distcut</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – Maximum distance between cell centers in consecutive frames for cells to be considered the same.</p></li>
<li><p><strong>framewindow</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Number of frames over which to look for cell correspondences.</p></li>
<li><p><strong>visual_1cell</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True, displays visual tracking information for single cell matches during processing.</p></li>
<li><p><strong>visual</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True, displays visual tracking information for all cells during processing.</p></li>
<li><p><strong>max_search_radius</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The maximum search radius in pixels for linking objects between frames.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True, saves the lineage data (<cite>linSet</cite>) to an HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>) – If True and save_h5 is True, overwrites existing data in the HDF5 file.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>linSet</strong> – A list of arrays where each array corresponds to a frame and contains indices that map
each cell to its predecessor in the previous frame. Cells with no predecessor are marked
with -1. The data saved in <cite>linSet</cite> thus represents the lineage of each cell over the stack.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>AttributeError</strong> – If <cite>tf_matrix_set</cite> is not set, indicating that stack transformation matrices are required
    for tracking but have not been calculated.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">lineage_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_lineage_btrack</span><span class="p">(</span><span class="n">mskchannel</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">visual</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_lineage_mindist">
<span class="sig-name descname"><span class="pre">get_lineage_mindist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">distcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visual</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_lineage_mindist"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_lineage_mindist" title="Link to this definition"></a></dt>
<dd><p>Tracks cell lineage based on the minimum distance between cells across consecutive frames.
This method assesses cell positions to establish lineage by identifying the nearest cell
in the subsequent frame within a specified distance threshold.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>distcut</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The maximum distance a cell can move between frames to be considered the same cell.
Cells moving a distance greater than this threshold will not be tracked from one frame to the next.</p></li>
<li><p><strong>visual</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, displays a visual representation of the tracking process for each frame, showing
the cells and their movements between frames.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the lineage data (<cite>linSet</cite>) to an HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is True, overwrites existing data in the HDF5 file.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>linSet</strong> – A list where each entry corresponds to a frame and contains cell indices that map each cell
to its predecessor in the previous frame. Cells with no identifiable predecessor are marked with -1.
This list provides a complete lineage map of cells across all analyzed frames.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>AttributeError</strong> – If the cell positions (<cite>x</cite>) are not calculated prior to running this method, indicating that
    <cite>get_cell_positions</cite> needs to be executed first.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lineage_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_lineage_mindist</span><span class="p">(</span><span class="n">distcut</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">visual</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_mask_data">
<span class="sig-name descname"><span class="pre">get_mask_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_frame</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_mask_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_mask_data" title="Link to this definition"></a></dt>
<dd><p>Retrieve the mask data for a specified frame from the HDF5 file associated with this instance.
This method accesses the HDF5 file, navigates to the specific dataset for the given frame,
and extracts the mask data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>n_frame</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The frame number from which to retrieve mask data.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>msk</strong> – The mask data as a NumPy array. The structure of the array will depend on the mask setup
in the HDF5 file, such as whether it includes dimensions for multiple channels or z-stacks.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mask_data</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_mask_data</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mask_data</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1024, 1024, 2)  # Example shape, actual may vary.</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_motility_features">
<span class="sig-name descname"><span class="pre">get_motility_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">indcells</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_motility_features"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_motility_features" title="Link to this definition"></a></dt>
<dd><p>Extracts motility features for individual cells and their neighbors. This method calculates
both single-cell and neighbor-averaged motility characteristics, such as displacement and
interaction with neighboring cells, based on tracking data and cell label information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>indcells</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of cells for which to calculate motility features. If None, features are calculated
for all cells in the dataset.</p></li>
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Mask channel used to define cell labels.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the calculated motility features to an HDF5 file specified in the trajectory object.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is True, overwrites existing data in the HDF5 file.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>Xf_com</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – An array of computed motility features for each specified cell. The array dimensions are
(number of cells, number of features).</p></li>
<li><p><strong>feature_list</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <em>optional</em>) – Descriptions of each motility feature computed. This is returned if <cite>return_feature_list</cite>
is set to True in the method call.</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If required data such as mask channels or cell indices are not set, or if cell tracking has
    not been performed prior to feature extraction.</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">motility_features</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_motility_features</span><span class="p">(</span><span class="n">indcells</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">],</span> <span class="n">mskchannel</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_pair_rdf">
<span class="sig-name descname"><span class="pre">get_pair_rdf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cell_indsA</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cell_indsB</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rbins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_pair_rdf"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_pair_rdf" title="Link to this definition"></a></dt>
<dd><p>Calculates the radial distribution function (RDF) between two sets of cells, identifying the
frequency of cell-cell distances within specified radial bins. This method is commonly used
in statistical physics and materials science to study the spatial distribution of particles.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cell_indsA</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of the first set of cells. If None, considers all cells.</p></li>
<li><p><strong>cell_indsB</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of the second set of cells. If None, uses the same indices as <cite>cell_indsA</cite>.</p></li>
<li><p><strong>rbins</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – Array of radial bins for calculating RDF. If None, bins are generated linearly from nearly 0 to <cite>rmax</cite>.</p></li>
<li><p><strong>nr</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Number of radial bins if <cite>rbins</cite> is not specified. Default is 50.</p></li>
<li><p><strong>rmax</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – Maximum radius for the radial bins if <cite>rbins</cite> is not specified. Default is 500 units.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>rbins</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – The radial bins used for the RDF calculation, adjusted to remove the zero point and ensure proper binning.</p></li>
<li><p><strong>paircorrx</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – RDF values corresponding to each radial bin, normalized to the total number of pairs and the bin volumes.</p></li>
</ul>
</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rbins</span><span class="p">,</span> <span class="n">rdf</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_pair_rdf</span><span class="p">(</span><span class="n">cell_indsA</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">],</span> <span class="n">cell_indsB</span><span class="o">=</span><span class="p">[</span><span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">],</span> <span class="n">nr</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">rmax</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
<span class="go"># This will calculate the RDF between two specified sets of cells with 100 radial bins up to a maximum radius of 200.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The RDF gives a normalized measure of how often pairs of points (cells) appear at certain distances from each other,</p></li>
</ul>
<p>compared to what would be expected for a completely random distribution at the same density.
- This function is useful for examining the spatial organization and clustering behavior of cells in tissues or cultures.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_secreted_ligand_density">
<span class="sig-name descname"><span class="pre">get_secreted_ligand_density</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">frame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mskchannel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">npad</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">indz_bm</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">secretion_rate</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">D</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">flipz</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visual</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_secreted_ligand_density"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_secreted_ligand_density" title="Link to this definition"></a></dt>
<dd><p>Simulates the diffusion of secreted ligands from cells, providing a spatial distribution of ligand density across a specified frame.
The simulation considers specified boundary conditions and secretion rates to model the ligand concentration in the vicinity of cells.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>frame</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The frame index from which image and mask data are extracted.</p></li>
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The channel of the mask that identifies the cells.</p></li>
<li><p><strong>scale</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The scaling factor for the resolution of the simulation. Default is 2.0.</p></li>
<li><p><strong>npad</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array-like</span></code>, <em>optional</em>) – Padding to add around the simulation area to avoid edge effects. Defaults to [0, 0, 0] if None.</p></li>
<li><p><strong>indz_bm</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The index of the bottom-most slice to consider in the z-dimension.</p></li>
<li><p><strong>secretion_rate</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">array-like</span></code>, <em>optional</em>) – The rate at which ligands are secreted by the cells. Can be a single value or an array specifying different rates for different cells.</p></li>
<li><p><strong>D</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – Diffusion coefficient. If not specified, it is calculated based on the pixel size and z-scaling.</p></li>
<li><p><strong>flipz</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, flips the z-dimension of the image and mask data, useful for certain imaging orientations.</p></li>
<li><p><strong>visual</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, displays visualizations of the simulation process and results.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>vdist</strong> – A 3D array representing the volumetric distribution of the ligand density around cells in the specified frame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">ligand_density</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_secreted_ligand_density</span><span class="p">(</span><span class="n">frame</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">mskchannel</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">scale</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span> <span class="n">secretion_rate</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">D</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>
<span class="go"># This will simulate and return the ligand density around cells in frame 10 with specified parameters.</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If any of the provided indices or parameters are out of the expected range or if there is a mismatch in array dimensions.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The method performs a complex series of image processing steps including scaling, padding, flipping, and 3D mesh generation.</p></li>
<li><p>It uses finite element methods to solve diffusion equations over the generated mesh, constrained by the cellular boundaries and secretion rates.</p></li>
</ul>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_signal_contributions">
<span class="sig-name descname"><span class="pre">get_signal_contributions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">S</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">time_lag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_pos</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">R</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zscale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rescale_z</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_signal_contributions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_signal_contributions" title="Link to this definition"></a></dt>
<dd><p>Computes the spatial contributions of signaling between cells over a specified time lag. This method
averages signals from nearby cells, weighted inversely by their distances, to assess local signaling interactions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>S</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – A binary array indicating the signaling status of cells (1 for active, 0 for inactive).</p></li>
<li><p><strong>time_lag</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The time lag over which to assess signal contributions, defaulting to 0 for immediate interactions.</p></li>
<li><p><strong>x_pos</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – Positions of cells. If None, the positions are taken from the instance’s <cite>x</cite> attribute.</p></li>
<li><p><strong>rmax</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The maximum radius within which to consider signal contributions from neighboring cells, default is 5.</p></li>
<li><p><strong>R</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – Normalization radius, typically set to the average cell diameter; defaults to the instance’s <cite>cellpose_diam</cite>.</p></li>
<li><p><strong>zscale</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The scaling factor for the z-dimension, used if <cite>rescale_z</cite> is True.</p></li>
<li><p><strong>rescale_z</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, scales the z-coordinates of positions by <cite>zscale</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>S_r</strong> – An array where each element is the averaged spatial signal contribution received by each cell, normalized
by distance and weighted by the signaling status of neighboring cells.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">S</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">randint</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">traj</span><span class="o">.</span><span class="n">cells_indSet</span><span class="o">.</span><span class="n">size</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">signal_contributions</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_signal_contributions</span><span class="p">(</span><span class="n">S</span><span class="p">,</span> <span class="n">time_lag</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">rmax</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">R</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>
<span class="go"># This computes the signal contributions for each cell, considering interactions within a radius of 10 units.</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>This method is useful for understanding the influence of cell-cell interactions within a defined spatial range</p></li>
</ul>
<p>and can be particularly insightful in dynamic cellular environments where signaling is a key factor.
- The distances are normalized by the cell radius <cite>R</cite> to provide a relative measure of proximity, and the contributions
are weighted by the inverse of these normalized distances.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If necessary parameters are missing or incorrectly formatted.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_stack_trans">
<span class="sig-name descname"><span class="pre">get_stack_trans</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mskchannel=0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ntrans=20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxt=10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_function=&lt;function</span> <span class="pre">get_pairwise_distance_sum&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zscale=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_h5=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_global=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">**dist_function_keys</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_stack_trans"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_stack_trans" title="Link to this definition"></a></dt>
<dd><p>Computes translations across an image stack using a brute force optimization method to align
cell centers from frame to frame. This method can apply both local and global alignment strategies
based on the distribution of cell centers.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mskchannel</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – Mask channel to use for extracting cell centers from labels.</p></li>
<li><p><strong>ntrans</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Number of translations to try in each dimension during optimization.</p></li>
<li><p><strong>maxt</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Maximum translation distance to consider in each dimension.</p></li>
<li><p><strong>dist_function</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">function</span></code>) – Optimization function that takes cell centers from two frames and a translation vector,
returning a score where lower values indicate better alignment.</p></li>
<li><p><strong>zscale</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – Scaling factor for the z-dimension to normalize it with x and y dimensions.</p></li>
<li><p><strong>save_h5</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, saves the computed transformation matrices to an HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True and save_h5 is True, overwrites existing data in the HDF5 file.</p></li>
<li><p><strong>do_global</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, performs a global alignment using the center of mass of all masks prior to brute force optimization.</p></li>
<li><p><strong>dist_function_keys</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">dict</span></code>) – Additional keyword arguments to pass to the dist_function.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>tf_matrix_set</strong> – An array of shape (nframes, ndim+1, ndim+1) containing the transformation matrices for aligning
each frame to the first frame based on the computed translations.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">transformations</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_stack_trans</span><span class="p">(</span><span class="n">mskchannel</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ntrans</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">maxt</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">do_global</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_trajAB_segments">
<span class="sig-name descname"><span class="pre">get_trajAB_segments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">xt</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stateA</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stateB</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clusters</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distcutA</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distcutB</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_trajAB_segments"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_trajAB_segments" title="Link to this definition"></a></dt>
<dd><p>Identifies segments within trajectories that transition between specified states, A and B. This method
can be used to analyze transitions or dwell times in specific states within a trajectory dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>xt</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – An array representing trajectories, either as direct state assignments or continuous data.</p></li>
<li><p><strong>stateA</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">array-like</span></code>, <em>optional</em>) – The state or states considered as ‘A’. Transitions from this state are analyzed.</p></li>
<li><p><strong>stateB</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code> or <code class="xref py py-class docutils literal notranslate"><span class="pre">array-like</span></code>, <em>optional</em>) – The state or states considered as ‘B’. If defined, transitions from state A to state B are analyzed.</p></li>
<li><p><strong>clusters</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code>, <em>optional</em>) – A clustering object with an ‘assign’ method that can be used to discretize continuous trajectory data into states.</p></li>
<li><p><strong>states</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – An array defining all possible states. Used to map states in ‘xt’ if it contains direct state assignments.</p></li>
<li><p><strong>distcutA</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The distance cutoff for determining membership in state A if ‘xt’ is continuous.</p></li>
<li><p><strong>distcutB</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>, <em>optional</em>) – The distance cutoff for determining membership in state B if ‘xt’ is continuous and ‘stateB’ is defined.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>slices</strong> – A list of slice objects representing the indices of ‘xt’ where transitions between specified states occur.
If only ‘stateA’ is specified, returns segments where the trajectory is in state A.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">slice</span></code></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If required parameters for defining states or transitions are not provided or if the provided
    parameters are incompatible (e.g., ‘distcutA’ without a corresponding ‘stateA’).</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">xt</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">rand</span><span class="p">(</span><span class="mi">100</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>  <span class="c1"># Example continuous trajectory data</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">clusters</span> <span class="o">=</span> <span class="n">KMeans</span><span class="p">(</span><span class="n">n_clusters</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">fit</span><span class="p">(</span><span class="n">xt</span><span class="p">)</span>  <span class="c1"># Example clustering model</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">segments</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_trajAB_segments</span><span class="p">(</span><span class="n">xt</span><span class="p">,</span> <span class="n">stateA</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">stateB</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">clusters</span><span class="o">=</span><span class="n">clusters</span><span class="p">)</span>
<span class="go"># Analyze transitions from state 0 to state 1 using cluster assignments</span>
</pre></div>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>If ‘xt’ contains direct state assignments, ‘states’ must be provided to map these to actual state values.</p></li>
<li><p>For continuous data, ‘clusters’ or distance cutoffs (‘distcutA’, ‘distcutB’) must be used to define states.</p></li>
<li><p>This function is useful for analyzing kinetic data where transitions between states are of interest.</p></li>
</ul>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_traj_segments">
<span class="sig-name descname"><span class="pre">get_traj_segments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seg_length</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_traj_segments"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_traj_segments" title="Link to this definition"></a></dt>
<dd><p>Divides each trajectory into multiple overlapping segments of a specified length. This method
is useful for analyzing sections of trajectories or for preparing data for machine learning
models that require fixed-size input.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>seg_length</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – The length of each segment to be extracted from the trajectories. Segments are created
by sliding a window of this length along each trajectory.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>traj_segSet</strong> – A 2D array where each row represents a segment of a trajectory. The number of columns
in this array equals <cite>seg_length</cite>. Each segment includes consecutive cell indices
from the original trajectories.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>This method requires that the <cite>trajectories</cite> attribute has been populated, typically by</p></li>
</ul>
<p>a method that computes full trajectories such as <cite>get_unique_trajectories</cite>.</p>
<ul class="simple">
<li><p>Only trajectories that are at least as long as <cite>seg_length</cite> will contribute segments to</p></li>
</ul>
<p>the output. Shorter trajectories are ignored.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">get_unique_trajectories</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">segments</span> <span class="o">=</span> <span class="n">traj</span><span class="o">.</span><span class="n">get_traj_segments</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">segments</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>
<span class="go">(number of segments, 5)  # Example shape, actual values depend on trajectory lengths and seg_length.</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – If <cite>seg_length</cite> is larger than the length of any available trajectory, resulting in
    no valid segments being produced.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_trajectory_steps">
<span class="sig-name descname"><span class="pre">get_trajectory_steps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">inds</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">traj</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Xtraj</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_trajectories</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nlag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_trajectory_steps"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_trajectory_steps" title="Link to this definition"></a></dt>
<dd><p>Extracts sequential steps from cell trajectories and retrieves corresponding features from a feature matrix.
This method is useful for analyses that require step-wise comparison of trajectories, such as
calculating changes or transitions over time.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>inds</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Indices of cells for which to get trajectory steps. If None, processes all cells.</p></li>
<li><p><strong>traj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – The trajectory data matrix. If None, uses the instance’s <cite>traj</cite> attribute.</p></li>
<li><p><strong>Xtraj</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <em>optional</em>) – The feature data matrix corresponding to trajectories. If None, uses the instance’s <cite>Xtraj</cite> attribute.</p></li>
<li><p><strong>get_trajectories</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, computes unique trajectories for the specified indices before processing steps.</p></li>
<li><p><strong>nlag</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – The lag between steps in a trajectory to consider. A value of 1 means consecutive steps.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The method assumes that the trajectory and feature data matrices (<cite>traj</cite> and <cite>Xtraj</cite>, respectively)</p></li>
</ul>
<p>are indexed in the same way.
- This function can optionally calculate unique trajectories before extracting steps, making it
versatile for both freshly calculated and pre-computed trajectory datasets.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/data.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">get_trajectory_steps</span><span class="p">(</span><span class="n">get_trajectories</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">nlag</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="go"># This will compute unique trajectories for all cells and then extract every second step.</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>IndexError</strong> – If any index in <cite>inds</cite> is out of bounds of the available data.</p></li>
<li><p><strong>ValueError</strong> – If <cite>traj</cite> or <cite>Xtraj</cite> data matrices are not set and not provided as arguments.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.get_unique_trajectories">
<span class="sig-name descname"><span class="pre">get_unique_trajectories</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cell_inds</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extra_depth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.get_unique_trajectories"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.get_unique_trajectories" title="Link to this definition"></a></dt>
<dd><p>Computes unique trajectories for a set of cells over multiple frames, minimizing redundancy by
ensuring that no two trajectories cover the same cell path beyond a specified overlap (extra_depth).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cell_inds</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">array</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Array of cell indices for which to calculate trajectories. If None, calculates trajectories
for all cells.</p></li>
<li><p><strong>verbose</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, provides detailed logs during the trajectory calculation process.</p></li>
<li><p><strong>extra_depth</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>, <em>optional</em>) – Specifies how many frames of overlap to allow between different trajectories. If not set,
uses the pre-set attribute ‘trajl’ minus one as the depth; if ‘trajl’ is not set, defaults to 0.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>This method identifies unique trajectories by tracking each cell backward from its last appearance</p></li>
</ul>
<p>to its first, recording the trajectory, and then ensuring subsequent trajectories do not retread
the same path beyond the allowed overlap specified by ‘extra_depth’.</p>
<ul class="simple">
<li><p>Each trajectory is tracked until it either reaches the start of the dataset or an earlier part of</p></li>
</ul>
<p>another trajectory within the allowed overlap.</p>
<ul class="simple">
<li><p>This function updates the instance’s ‘trajectories’ attribute, storing each unique trajectory.</p></li>
</ul>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">get_unique_trajectories</span><span class="p">(</span><span class="n">verbose</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.load_from_h5">
<span class="sig-name descname"><span class="pre">load_from_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.load_from_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.load_from_h5" title="Link to this definition"></a></dt>
<dd><p>Load data from a specified path within an HDF5 file. This method attempts to read records
recursively from the given path in the HDF5 file specified by the <cite>h5filename</cite> attribute
of the instance.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>path</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – The base path in the HDF5 file from which to load data.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Returns True if the data was successfully loaded, False otherwise.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">load_from_h5</span><span class="p">(</span><span class="s1">&#39;/data/group1&#39;</span><span class="p">)</span>
<span class="go">loading path/to/your/hdf5file.h5</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="celltraj.trajectory.Trajectory.save_to_h5">
<span class="sig-name descname"><span class="pre">save_to_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">attribute_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/trajectory.html#Trajectory.save_to_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.trajectory.Trajectory.save_to_h5" title="Link to this definition"></a></dt>
<dd><p>Save specified attributes to an HDF5 file at the given path. This method saves attributes from
the current instance to a specified location within the HDF5 file, creating or overwriting data
as necessary based on the <cite>overwrite</cite> parameter.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – The base path in the HDF5 file where attributes will be saved.</p></li>
<li><p><strong>attribute_list</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code> of <code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>) – A list containing the names of attributes to save to the HDF5 file.</p></li>
<li><p><strong>overwrite</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code>, <em>optional</em>) – If True, existing data at the specified path will be overwritten. Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Returns True if the attributes were successfully saved, False otherwise, such as when the
HDF5 file does not exist or attributes cannot be written.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">bool</span></code></p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span> <span class="o">=</span> <span class="n">Trajectory</span><span class="p">(</span><span class="s1">&#39;path/to/your/hdf5file.h5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">some_attribute</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">traj</span><span class="o">.</span><span class="n">save_to_h5</span><span class="p">(</span><span class="s1">&#39;/data/&#39;</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;some_attribute&#39;</span><span class="p">])</span>
<span class="go">saving attributes [&#39;some_attribute&#39;] to /data/ in path/to/your/hdf5file.h5</span>
<span class="go">saved some_attribute to path/to/your/hdf5file.h5/data/</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

</section>
<section id="module-celltraj.features">
<span id="celltraj-features"></span><h2>celltraj.features<a class="headerlink" href="#module-celltraj.features" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.apply3d">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">apply3d</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">function2d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dtype</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">function2d_args</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#apply3d"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.apply3d" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.boundaryCB_FFT">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">boundaryCB_FFT</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">msk</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmsk</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncomp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">15</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">256</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bordersize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#boundaryCB_FFT"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.boundaryCB_FFT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.boundaryFFT">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">boundaryFFT</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">msk</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncomp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">15</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">256</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#boundaryFFT"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.boundaryFFT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featBoundary">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featBoundary</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featBoundary"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featBoundary" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featBoundaryCB">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featBoundaryCB</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featBoundaryCB"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featBoundaryCB" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featHaralick">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featHaralick</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featHaralick"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featHaralick" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featNucBoundary">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featNucBoundary</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featNucBoundary"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featNucBoundary" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featSize">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featSize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featSize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featSize" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.featZernike">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">featZernike</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#featZernike"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.featZernike" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.get_cc_cs_border">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">get_cc_cs_border</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mskcell</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmskcell</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bordersize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#get_cc_cs_border"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.get_cc_cs_border" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.get_contact_boundaries">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">get_contact_boundaries</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary_only</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#get_contact_boundaries"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.get_contact_boundaries" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.get_contact_labels">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">get_contact_labels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#get_contact_labels"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.get_contact_labels" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.get_neighbor_feature_map">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">get_neighbor_feature_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">neighbor_function=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contact_labels=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dtype=&lt;class</span> <span class="pre">'numpy.float64'&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">**neighbor_function_args</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#get_neighbor_feature_map"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.get_neighbor_feature_map" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.get_pca_fromdata">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">get_pca_fromdata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">var_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.95</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#get_pca_fromdata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.get_pca_fromdata" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.meanIntensity">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">meanIntensity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#meanIntensity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.meanIntensity" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.features.totalIntensity">
<span class="sig-prename descclassname"><span class="pre">celltraj.features.</span></span><span class="sig-name descname"><span class="pre">totalIntensity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regionmask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intensity</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/features.html#totalIntensity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.features.totalIntensity" title="Link to this definition"></a></dt>
<dd></dd></dl>

</section>
<section id="module-celltraj.imageprep">
<span id="celltraj-imageprep"></span><h2>celltraj.imageprep<a class="headerlink" href="#module-celltraj.imageprep" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.clean_labeled_mask">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">clean_labeled_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">masks_nuc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_borders</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_padding</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fill_holes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'largest'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">test_map</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">test_cut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#clean_labeled_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.clean_labeled_mask" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.create_h5">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">create_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dic</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#create_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.create_h5" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.crop_image">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">crop_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tshift</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nx</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ny</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#crop_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.crop_image" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.dist_to_contact">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">dist_to_contact</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">m</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">12</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#dist_to_contact"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.dist_to_contact" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.expand_registered_images">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">expand_registered_images</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">imgs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tSet</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#expand_registered_images"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.expand_registered_images" title="Link to this definition"></a></dt>
<dd><p>Apply transformations to a stack of images and expand images so they align
:param imgs: images (Z,X,Y), registration along Z tSet: transformations for each image (angle, x-trans, y-trans)
:type imgs: ndarray or list of images (each image same size) tSet: ndarray (NZ, 3)
:return: expanded and registered image stack
:rtype: ndarray (NZ, X, Y)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_cell_centers">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_cell_centers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_cell_centers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_cell_centers" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_cell_intensities">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_cell_intensities</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">averaging</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_3D</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_cell_intensities"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_cell_intensities" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_contactsum_dev">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_contactsum_dev</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">centers1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">img2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rp1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nt</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefile</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_contactsum_dev"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_contactsum_dev" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_cyto_minus_nuc_labels">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_cyto_minus_nuc_labels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels_cyto</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">labels_nuc</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_cyto_minus_nuc_labels"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_cyto_minus_nuc_labels" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_feature_map">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_feature_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">features</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">labels</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_feature_map"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_feature_map" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_images">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_images</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filelist</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_images"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_images" title="Link to this definition"></a></dt>
<dd><p>Get images from list of files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>filelist</strong> (<em>list</em><em> of </em><em>strings</em>) – list of image files</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>list of images</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of arrays</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_intensity_centers">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_intensity_centers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">msk</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">footprint_shape</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smooth_sigma</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pad_zeros</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_intensity_centers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_intensity_centers" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_label_largestcc">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_label_largestcc</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">label</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fill_holes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_label_largestcc"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_label_largestcc" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_labeled_mask">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_labeled_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">b_imgr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgM</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apply_watershed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fill_holes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_footprint</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zscale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_labeled_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_labeled_mask" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_mask_2channel_ilastik">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_mask_2channel_ilastik</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_ilastik</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fore_channel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holefill_area</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">growthcycles</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.8</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_mask_2channel_ilastik"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_mask_2channel_ilastik" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_masks">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_masks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">masklist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fore_channel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holefill_area</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">growthcycles</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.8</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_masks"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_masks" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_nndist_sum">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_nndist_sum</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tshift</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_nndist_sum"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_nndist_sum" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_pair_rdf_fromcenters">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_pair_rdf_fromcenters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rbins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_pair_rdf_fromcenters"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_pair_rdf_fromcenters" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_registration_expansions">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_registration_expansions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tf_matrix_set</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">imgdims</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_registration_expansions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_registration_expansions" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_registrations">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_registrations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">imgs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_registrations"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_registrations" title="Link to this definition"></a></dt>
<dd><p>Apply pystackreg to get registrations along image stack
:param imgs: images (Z,X,Y), registration along Z
:type imgs: ndarray
:return: set of transformations to register image stack, with the triplet (radial angle, x-translation, y-translation) for each image
:rtype: ndarray (NZ,3), NZ number of images along Z</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_slide_image">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_slide_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">imgs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nrows</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncols</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_inds</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">foverlap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">histnorm</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_slide_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_slide_image" title="Link to this definition"></a></dt>
<dd><p>Construct slide image from a set of tiles (fields of view).
Ordering from (get_tile_order).
:param imgs: list of images nrows: number of rows, default assumes a square tiling (64 images = 8 rows x 8 cols)
ncols: number of columns foverlap: fraction of overlap between images
:type imgs: list of 2D images (2D arrays) nrows: int ncols: int foverlap: float
:return: constructed slide image from image tiles
:rtype: 2D array</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_tf_matrix_2d">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_tf_matrix_2d</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_tf_matrix_2d"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_tf_matrix_2d" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_tile_order">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_tile_order</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">nrows</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncols</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snake</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_tile_order"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_tile_order" title="Link to this definition"></a></dt>
<dd><p>Construct ordering for to put together image tiles compatible with incell microscope.
:param nrows: number of rows ncols: number of columns
snake: snake across whole image (left to right, right to left, left to right…)
:type nrows: int ncols: int ncols: int snake: bool
:return: constructed 2D array of image indices
:rtype: 2D array (int)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_voronoi_masks">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_voronoi_masks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgM</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_voronoi_masks"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_voronoi_masks" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.get_voronoi_masks_fromcenters">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">get_voronoi_masks_fromcenters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">nuc_centers</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">imgM</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'closest'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#get_voronoi_masks_fromcenters"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.get_voronoi_masks_fromcenters" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.histogram_stretch">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">histogram_stretch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">99</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#histogram_stretch"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.histogram_stretch" title="Link to this definition"></a></dt>
<dd><p>Histogram stretch of an array or image for normalization..</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>img</strong> – array or image</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>histogram stretched array</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>real array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.list_images">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">list_images</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">imagespecifier</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#list_images"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.list_images" title="Link to this definition"></a></dt>
<dd><p>list images in a directory matching a pattern..</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>imagepath</strong> (<em>string filespecifier string</em>) – directory :param filespecifier pattern to match for image files</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>list of matching imagefiles</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of strings</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.load_for_viewing">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">load_for_viewing</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fname</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#load_for_viewing"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.load_for_viewing" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.load_ilastik">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">load_ilastik</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_ilastik</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#load_ilastik"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.load_ilastik" title="Link to this definition"></a></dt>
<dd><p>Load ilastik prediction (pixel classification) from h5 file format.
:param file_ilastik: filename
:type file_ilastik: string
:return: ndarray of ilastik output
:rtype: 2Dxn array (2D image by n ilastik labels)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.local_threshold">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">local_threshold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">imgr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">block_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">51</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">z_std</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#local_threshold"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.local_threshold" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.make_odd">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">make_odd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#make_odd"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.make_odd" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.organize_filelist_fov">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">organize_filelist_fov</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filelist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fov_pos</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fov_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#organize_filelist_fov"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.organize_filelist_fov" title="Link to this definition"></a></dt>
<dd><p>Organize imagefiles in a list to field of view.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>filelist</strong> (<em>list</em><em> of </em><em>strings fov_pos: int fov_len: int</em>) – list of image files :param fov_pos: string position of fov specifier :param fov_len: length of fov speficier</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>list of imagefiles organized by fov (increasing)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of strings</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.organize_filelist_time">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">organize_filelist_time</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filelist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">time_pos</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#organize_filelist_time"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.organize_filelist_time" title="Link to this definition"></a></dt>
<dd><p>Organize imagefiles in a list to timestamp ??d??h??m.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>filelist</strong> (<em>list</em><em> of </em><em>strings fov_pos: int fov_len: int</em>) – list of image files</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>list of imagefiles organized by fov (increasing)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of strings</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.pad_image">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">pad_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">maxdims</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">padvalue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#pad_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.pad_image" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.save_for_viewing">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">save_for_viewing</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metadata</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#save_for_viewing"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.save_for_viewing" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.save_frame_h5">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">save_frame_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">frame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">img</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">msks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fmsk</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">features</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">overwrite</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">timestamp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#save_frame_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.save_frame_h5" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.transform_image">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">transform_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tf_matrix</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse_tform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pad_dims</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">ndimage_args</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#transform_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.transform_image" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.imageprep.znorm">
<span class="sig-prename descclassname"><span class="pre">celltraj.imageprep.</span></span><span class="sig-name descname"><span class="pre">znorm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/imageprep.html#znorm"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.imageprep.znorm" title="Link to this definition"></a></dt>
<dd><p>Variance normalization (z-norm) of an array or image)..</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>img</strong> – array or image</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>z-normed array</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>real array</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-celltraj.model">
<span id="celltraj-model"></span><h2>celltraj.model<a class="headerlink" href="#module-celltraj.model" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.clean_clusters">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">clean_clusters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">clusters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">P</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#clean_clusters"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.clean_clusters" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_H_eigs">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_H_eigs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mt</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_H_eigs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_H_eigs" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_avdx_clusters">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_avdx_clusters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x_clusters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Mt</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_avdx_clusters"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_avdx_clusters" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_committor">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_committor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Tmatrix</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">indTargets</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">indSource</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conv</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_committor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_committor" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_dmat">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_dmat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_dmat"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_dmat" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_gaussianKernelM">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_gaussianKernelM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">M</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_gaussianKernelM"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_gaussianKernelM" title="Link to this definition"></a></dt>
<dd><p>Get Malanobis scaled gaussian kernel from observation matrices X,Y</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix, samples x features</p></li>
<li><p><strong>Y</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix at t+1, samples x features</p></li>
<li><p><strong>M</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Mahalanobis scaling matrix, features x features</p></li>
<li><p><strong>h</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – vector of sigma scalings for gaussian from get_kernel_sigmas</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>k</strong> – Malanobis scaled kernel for X,Y</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">samples</span> <span class="pre">x</span> <span class="pre">samples</span> <span class="pre">(float)</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_kernel_sigmas">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_kernel_sigmas</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">M</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vector_sigma</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_kernel_sigmas"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_kernel_sigmas" title="Link to this definition"></a></dt>
<dd><p>Get sigmas from observation matrix</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix, samples x features</p></li>
<li><p><strong>M</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Mahalanobis scaling matrix, features x features</p></li>
<li><p><strong>s</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – bandwidth scaling factor</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>h</strong> – vector of sigmas to scale observations in kernel</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">n_features</span> <span class="pre">(float)</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_kineticstates">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_kineticstates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vkin</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nstates_final</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nstates_initial</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pcut_final</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_nstates_initial</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_ninit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_kineticstates"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_kineticstates" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_koopman_eig">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_koopman_eig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">M</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1e-05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">psi_X</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">psi_Y</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_koopman_eig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_koopman_eig" title="Link to this definition"></a></dt>
<dd><p>Get linear matrix solution for Koopman operator from X,Y paired observation Y=F(X) with F the forward operator</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix, samples x features</p></li>
<li><p><strong>Y</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix at t+1, samples x features</p></li>
<li><p><strong>M</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Mahalanobis scaling matrix, features x features</p></li>
<li><p><strong>s</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – kernel bandwidth scaling parameter</p></li>
<li><p><strong>bta</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code>) – regularization parameter for linear solve</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>K</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman operator matrix, samples x samples (float)</p></li>
<li><p><strong>Xi</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman left eigenvectors, samples x samples</p></li>
<li><p><strong>Lam</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenvalue matrix (diagonal), samples x samples</p></li>
<li><p><strong>W</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman right eigenvectors, samples x samples</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_koopman_inference">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_koopman_inference</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">steps</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">V</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Lam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nmodes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_koopman_inference"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_koopman_inference" title="Link to this definition"></a></dt>
<dd><p>Get Koopman prediction of an observable</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>start</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – sample index for start point</p></li>
<li><p><strong>steps</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – number of steps of inference to perform</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
<li><p><strong>V</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman modes of observables, must be calculated samples x samples</p></li>
<li><p><strong>Lam</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenvalues matrix (diagonal), samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>X_pred</strong> – predicted trajectory steps x observables</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_koopman_inference_multiple">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_koopman_inference_multiple</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">starts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">steps</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">V</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Lam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nmodes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_koopman_inference_multiple"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_koopman_inference_multiple" title="Link to this definition"></a></dt>
<dd><p>Get Koopman prediction of an observable</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>start</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – sample index for start point</p></li>
<li><p><strong>steps</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">int</span></code>) – number of steps of inference to perform</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
<li><p><strong>V</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman modes of observables, must be calculated samples x samples</p></li>
<li><p><strong>Lam</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenvalues matrix (diagonal), samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>X_pred</strong> – predicted trajectory steps x observables</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_koopman_modes">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_koopman_modes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">psi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">W</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X_obs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1e-05</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_koopman_modes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_koopman_modes" title="Link to this definition"></a></dt>
<dd><p>Get Koopman modes of an observable</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>psi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – kernel matrix, samples x samples</p></li>
<li><p><strong>Xi</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – right eigenvectors of Koopman, samples x samples</p></li>
<li><p><strong>W</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – left eigenvectors of Koopman, samples x samples</p></li>
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – observation matrix, samples x features</p></li>
<li><p><strong>X_obs</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">samples</span> <span class="pre">x</span> <span class="pre">observables</span></code>) – observables of interest in same time order as X, can be X or features of X</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions</p></li>
<li><p><strong>V</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman modes of observables</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_kscore">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_kscore</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mt</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_kscore"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_kscore" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_landscape_coords_umap">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_landscape_coords_umap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vkin</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_landscape_coords_umap"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_landscape_coords_umap" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_motifs">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_motifs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">v</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncomp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">w</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_motifs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_motifs" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_path_entropy_2point">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_path_entropy_2point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Mt</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clusters</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exclude_stays</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_path_entropy_2point"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_path_entropy_2point" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_path_ll_2point">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_path_ll_2point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exclude_stays</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_path_ll_2point"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_path_ll_2point" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_steady_state_matrixpowers">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_steady_state_matrixpowers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Tmatrix</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">conv</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_steady_state_matrixpowers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_steady_state_matrixpowers" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_traj_ll_gmean">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_traj_ll_gmean</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xt</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exclude_stays</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_traj_ll_gmean"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_traj_ll_gmean" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_transition_matrix">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_transition_matrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clusters</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_transition_matrix"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_transition_matrix" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.get_transition_matrix_CG">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">get_transition_matrix_CG</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clusters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">states</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#get_transition_matrix_CG"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.get_transition_matrix_CG" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.plot_dx_arrows">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">plot_dx_arrows</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x_clusters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dx_clusters</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#plot_dx_arrows"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.plot_dx_arrows" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.plot_eig">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">plot_eig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">v</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_clusters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncomp</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#plot_eig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.plot_eig" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.update_mahalanobis_matrix_J">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">update_mahalanobis_matrix_J</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mprev</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">V</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#update_mahalanobis_matrix_J"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.update_mahalanobis_matrix_J" title="Link to this definition"></a></dt>
<dd><p>Update estimation of mahalanobis matrix for kernel tuning</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Mprev</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">features</span> <span class="pre">x</span> <span class="pre">features</span></code>) – Koopman eigenfunctions</p></li>
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – samples by features</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>M</strong> – updated mahalanobis matrix using Koopman eigenfunction gradients</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.update_mahalanobis_matrix_J_old">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">update_mahalanobis_matrix_J_old</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mprev</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">V</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Lam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#update_mahalanobis_matrix_J_old"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.update_mahalanobis_matrix_J_old" title="Link to this definition"></a></dt>
<dd><p>Update estimation of mahalanobis matrix for kernel tuning</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Mprev</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">features</span> <span class="pre">x</span> <span class="pre">features</span></code>) – Koopman eigenfunctions</p></li>
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – samples by features</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>M</strong> – updated mahalanobis matrix using Koopman eigenfunction gradients</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.update_mahalanobis_matrix_flux">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">update_mahalanobis_matrix_flux</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mprev</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">V</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">Lam</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#update_mahalanobis_matrix_flux"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.update_mahalanobis_matrix_flux" title="Link to this definition"></a></dt>
<dd><p>Update estimation of mahalanobis matrix for kernel tuning</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Mprev</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">features</span> <span class="pre">x</span> <span class="pre">features</span></code>) – Koopman eigenfunctions</p></li>
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – samples by features</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>M</strong> – updated mahalanobis matrix using Koopman eigenfunction gradients</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.model.update_mahalanobis_matrix_grad">
<span class="sig-prename descclassname"><span class="pre">celltraj.model.</span></span><span class="sig-name descname"><span class="pre">update_mahalanobis_matrix_grad</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">Mprev</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi_X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/model.html#update_mahalanobis_matrix_grad"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.model.update_mahalanobis_matrix_grad" title="Link to this definition"></a></dt>
<dd><p>Update estimation of mahalanobis matrix for kernel tuning</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Mprev</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">features</span> <span class="pre">x</span> <span class="pre">features</span></code>) – Koopman eigenfunctions</p></li>
<li><p><strong>X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – samples by features</p></li>
<li><p><strong>phi_X</strong> (<code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code>) – Koopman eigenfunctions, samples x samples</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>M</strong> – updated mahalanobis matrix using Koopman eigenfunction gradients</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="xref py py-class docutils literal notranslate"><span class="pre">ndarray</span></code></p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-celltraj.translate">
<span id="celltraj-translate"></span><h2>celltraj.translate<a class="headerlink" href="#module-celltraj.translate" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="celltraj.translate.get_null_correlations">
<span class="sig-prename descclassname"><span class="pre">celltraj.translate.</span></span><span class="sig-name descname"><span class="pre">get_null_correlations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_fc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_fc_states</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_fc_predicted</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nrandom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nchunk</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_file</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/translate.html#get_null_correlations"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.translate.get_null_correlations" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.translate.get_predictedFC">
<span class="sig-prename descclassname"><span class="pre">celltraj.translate.</span></span><span class="sig-name descname"><span class="pre">get_predictedFC</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">state_probs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">statesFC</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/translate.html#get_predictedFC"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.translate.get_predictedFC" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.translate.get_state_decomposition">
<span class="sig-prename descclassname"><span class="pre">celltraj.translate.</span></span><span class="sig-name descname"><span class="pre">get_state_decomposition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_fc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">state_probs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">npermutations</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inds_tm_training</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_file</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visual</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nchunk</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lb</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ub</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/translate.html#get_state_decomposition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.translate.get_state_decomposition" title="Link to this definition"></a></dt>
<dd></dd></dl>

</section>
<section id="module-celltraj.utilities">
<span id="celltraj-utilities"></span><h2>celltraj.utilities<a class="headerlink" href="#module-celltraj.utilities" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.colorbar">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">colorbar</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mappable</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#colorbar"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.colorbar" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.dist">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">dist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">img1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">img2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#dist"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.dist" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.dist_to_contact">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">dist_to_contact</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">m</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">12</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#dist_to_contact"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.dist_to_contact" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_cdist">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_cdist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">prob1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_cdist"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_cdist" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_cdist2d">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_cdist2d</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">prob1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_cdist2d"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_cdist2d" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_cell_centers">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_cell_centers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_cell_centers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_cell_centers" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_dmat">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_dmat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_dmat"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_dmat" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_dmat_vectorized">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_dmat_vectorized</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x2</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_dmat_vectorized"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_dmat_vectorized" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_linear_batch_normalization">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_linear_batch_normalization</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feat0_all</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feat1_all</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nbins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_coef</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stdcut</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_linear_batch_normalization"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_linear_batch_normalization" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_linear_coef">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_linear_coef</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">countsA</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">countsB</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">countsAB</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_linear_coef"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_linear_coef" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_meshfunc_average">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_meshfunc_average</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">faceValues</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">faceCenters</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bins</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_meshfunc_average"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_meshfunc_average" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_pairwise_distance_sum">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_pairwise_distance_sum</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tshift</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contact_transform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">m</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">12</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_pairwise_distance_sum"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_pairwise_distance_sum" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.get_tshift">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">get_tshift</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">centers1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centers2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_function</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ntrans</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxt</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">dist_function_keys</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#get_tshift"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.get_tshift" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.load_dict_from_h5">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">load_dict_from_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#load_dict_from_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.load_dict_from_h5" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.recursively_load_dict_contents_from_group">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">recursively_load_dict_contents_from_group</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#recursively_load_dict_contents_from_group"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.recursively_load_dict_contents_from_group" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.recursively_save_dict_contents_to_group">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">recursively_save_dict_contents_to_group</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dic</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#recursively_save_dict_contents_to_group"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.recursively_save_dict_contents_to_group" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="celltraj.utilities.save_dict_to_h5">
<span class="sig-prename descclassname"><span class="pre">celltraj.utilities.</span></span><span class="sig-name descname"><span class="pre">save_dict_to_h5</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dic</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">h5file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/celltraj/utilities.html#save_dict_to_h5"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#celltraj.utilities.save_dict_to_h5" title="Link to this definition"></a></dt>
<dd></dd></dl>

</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="installation.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="stubs/celltraj.trajectory.html" class="btn btn-neutral float-right" title="celltraj.trajectory" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Jeremy Copperman.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>